debezium-core/src/main/java/io/debezium/function/BufferedBlockingConsumer.java

```diff
@@ -51,13 +52,13 @@ public default void flush() throws InterruptedException {
             private T last;
 
             @Override
-            public void accept(T t) throws InterruptedException {
+            public synchronized void accept(T t) throws InterruptedException {
```

Comment:
I'm concerned a bit about this, IIUC it'll enforce synchronization of all the producer threads which seems bad. I think we could use an `AtomicReference` instead. I'll push a PR for it and you can let me know what you think.

Reply:
@jpechane I've pushed an alternative here: https://github.com/debezium/debezium/pull/338. Could you check it out?